provider/internal/dockerfile.go

```diff
@@ -96,13 +97,22 @@ func (d *Dockerfile) validate(preview bool, c *Context) error {
 }
 
 func parseDockerfile(r io.Reader) error {
-	parsed, err := parser.Parse(r)
+	// Don't attempt to parse if this uses a custom syntax.
```

Comment:
would love this to say "Disable validation if this uses a custom syntax" but also moved down to the `if custom` or `DetectSyntax` statement.